bin/scpipe/lib/sc/pipe.rb

```diff
@@ -69,37 +71,32 @@ def prepare_pipe
       end
 
       def run_pipe
-        @@pipeproc = Proc.new {
-          trap("INT") do
-            Process.exit
-          end
-          rundir = Dir.pwd
+        rundir = Dir.pwd
+        while @done==false do
           IO.popen("cd #{rundir} && #{SC.sclang_path.chomp} -d #{rundir.chomp} -i scvim", "w") do |sclang|
-            loop {
-              x = `cat #{@@pipe_loc}`
-              sclang.print x if x
-            }
+            @f = File.open(@@pipe_loc, "r")
+            begin
+              while x = @f.read do 
```

Comment:
Another advantage of using `@f=File.open` is that execution blocks on reading `@f.read`, conveniently waiting for each block of code to inserted by `sc_dispatcher` before passing it on.

When `@f` is closed by the signal handlers below, `@f.read` throws an `IOError`, which is rescued and ignored.

If `HUP` was signalled, the while loop above restarts sclang. `INT` sets the done flag, so the program is allowed to exit naturally instead of calling exit.